warehouse/query-core/src/test/java/datawave/query/util/DateIndexIteratorTest.java

```diff
@@ -0,0 +1,247 @@
+package datawave.query.util;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+import java.util.BitSet;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import org.apache.accumulo.core.data.ArrayByteSequence;
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.iteratorsImpl.system.SortedMapIterator;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Joiner;
+
+import datawave.query.iterator.SourceManagerTest.MockIteratorEnvironment;
+
+/**
+ * Tests for the {@link DateIndexIterator}
+ */
+public class DateIndexIteratorTest {
+
+    private static final Logger log = LoggerFactory.getLogger(DateIndexIteratorTest.class);
+
+    private final SortedMap<Key,Value> data = new TreeMap<>();
+    private final Map<String,String> options = new HashMap<>();
+
+    private SortedKeyValueIterator<Key,Value> source;
+    private MockIteratorEnvironment env;
+
+    private int expectedNextCount = 0;
+    private int expectedSeekCount = 0;
+
+    @BeforeEach
+    public void setup() {
+        data.clear();
+        options.clear();
+        expectedNextCount = 0;
+        expectedSeekCount = 0;
+        env = new MockIteratorEnvironment();
+    }
+
+    @Test
+    public void testSimpleScan() throws IOException {
+        write("20250404", "LOADED", "20250403", "datatype-a", "FIELD_A", 4, 5, 6);
+        withExpectedNextSeekCounts(2, 1);
+        drive("20250404", "LOADED");
+    }
+
+    @Test
+    public void testMultipleLoadedDates() throws IOException {
+        write("20250404", "LOADED", "20250402", "datatype-a", "FIELD_A", 1, 2, 3);
+        write("20250404", "LOADED", "20250403", "datatype-a", "FIELD_A", 4, 5, 6);
+        write("20250404", "LOADED", "20250404", "datatype-a", "FIELD_A", 7, 8, 9);
+        withExpectedNextSeekCounts(4, 1);
+        drive("20250404", "LOADED");
```

Comment:
`drive()` is a bit ambiguous as a method name. Can this be renamed to `assertExpected()` or something similar?